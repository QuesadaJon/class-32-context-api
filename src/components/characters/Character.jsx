import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image }) => (
  <figure>
    <p>{name}</p>
    <img src={image} alt={name} />
  </figure>

);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Character;
