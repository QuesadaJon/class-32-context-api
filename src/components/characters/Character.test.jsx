import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';
import { MemoryRouter } from 'react-router-dom';

describe('displays an character', () => {
  it('renders one single character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Character 
          name="name" 
          image="image" 
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
