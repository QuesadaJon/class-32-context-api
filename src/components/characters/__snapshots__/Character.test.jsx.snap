// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an character renders one single character 1`] = `
<DocumentFragment>
  <figure>
    <p>
      name
    </p>
    <img
      alt="name"
      src="image"
    />
  </figure>
</DocumentFragment>
`;
